prime-router/src/main/kotlin/Translator.kt

```diff
@@ -49,27 +49,40 @@ class Translator(private val metadata: Metadata, private val settings: SettingsP
             receiver.topic == input.schema.topic &&
                 (limitReceiversTo.isEmpty() || limitReceiversTo.contains(receiver.fullName))
         }.mapNotNull { receiver ->
-            try {
-                // Filter the report
-                val filteredReport = filterByAllFilterTypes(settings, input, receiver) ?: return@mapNotNull null
-                if (filteredReport.isEmpty()) return@mapNotNull RoutedReport(filteredReport, receiver)
+            filterAndTranslateForReceiver(input, defaultValues, receiver, warnings)
+        }
+        return RoutedReportsResult(routedReports, warnings)
+    }
 
-                // Translate the filteredReport
-                val translatedReport = translateByReceiver(filteredReport, receiver, defaultValues)
-                RoutedReport(translatedReport, receiver)
-            } catch (e: IllegalStateException) {
-                // catching individual translation exceptions enables overall work to continue
-                warnings.add(
-                    ActionLog(
-                        InvalidTranslationMessage(e.localizedMessage),
-                        "TO:${receiver.fullName}:${receiver.schemaName}",
-                        reportId = input.id,
-                    )
+    /**
+     * Filter and translate for an [input] report for a single [receiver]. Use [defaultValues] and
+     * fill in [warnings].
+     */
+    fun filterAndTranslateForReceiver(
+        input: Report,
+        defaultValues: Map<String, String>,
+        receiver: Receiver,
+        warnings: MutableList<ActionLog>
```

Comment:
Rather than take in warnings, have it return any warnings. Should be able to reuse `RoutedReportResult`. Might result in a slight reshaping to  `filterAndTranslateByReceiver`

Reply:
Agree. Take a look at my latest version.